Added support for ASUS ProArt B760-CREATOR D4 (Nuvoton NCT6798D)

grigorievvs opened the pull request:
Basically, just added a specific motherboard, adjusted the fan names (CPU Opt, according to motherboard datasheet/instruction) and, most important, T Sensor.

PhyxionNL commented on LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:
```diff
@@ -4194,6 +4194,48 @@ private static void GetNuvotonConfigurationD(ISuperIO superIO, Manufacturer manu
+                        t.Add(new Temperature("Motherboard", 2)); // SYSTIN, MOTHERBOARD
+                        t.Add(new Temperature("T Sensor", 8)); // TSENSOR
+                        t.Add(new Temperature("PCH", 13)); // PCH_CHIP_TEMP
+                        t.Add(new Temperature("PECI_0_CAL", 22)); // PECI_0_CAL, CPU
```
Is there a better name for this?

grigorievvs replied:
1. I'm not quite sure about PECI_0_CAL (suppose, CAL means "Calculated" or even "Average"), but it definitely some CPU temp.

2. "T Sensor"/"TSENSOR" - well, according to motherboard specs, it is...temperature sensor, called by Asus as "Thermal Sensor Header".
Here is the link for their [support site](https://www.asus.com/motherboards-components/motherboards/proart/proart-b760-creator-d4/helpdesk_manual?model2Name=ProArt-B760-CREATOR-D4).
Direct link to motherboard [user's manual](https://dlcdnets.asus.com/pub/ASUS/mb/LGA1700/ProArt_B760-CREATOR_D4/E21433_ProArt_B760-CREATOR_D4_UM_WEB.pdf?model=ProArt%20B760-CREATOR%20D4)
_*Not trying to waste your time, just in case you want to see their naming._

3. "Motherboard"/"SYSTIN" seems to be typical motherboard sensor, but i'm not sure where it exactly placed.
Will try to find out and reply.

4. "PCH"/"PCH_CHIP_TEMP" - chipset temperature, i really don't know how to name it more accurate/correct.
Seems to me it's pretty self-explanatory.

--

If you can point me out to some naming guideline or "see this mobo, use it naming" - no problem.
I suppose "T Sensor" probably should be named as "Thermal Sensor Header", but it's Asus naming (or, let's say - extended naming), not common (afaik).

Or, if you can say "debug there, show output" or something like that - again, no problem.
I own this motherboard right now and can debug needed stuff.

--

Or do you mean voltages, like "Voltage #X"? 
I took them from another motherboard on same chip (NCT6798D), slightly checked to match other tools readings, but probably can try to match with BIOS/other tools, if it needed.

PhyxionNL replied:
A good name would be: `PECI 0 Calibrated`.
Also CPU Opt > CPU Optional.

Splenium38 commented:
MSI x870 Tomahawk also using this chip can i use this changes with my app to solve my problems with FanControl and if yes how ?

grigorievvs commented:
> MSI x870 Tomahawk also using this chip can i use this changes with my app to solve my problems with FanControl and if yes how ?

I will describe the whole path, but if you don't need to know "how" / already know "how to", just skip the unnecessary steps.

My way was:

1. Get other tools to check temps/fans.
I don’t know if it is allowed to mention it here, but I used AIDA, HWInfo, and HWMonitor.

2. Clone the project and open it in an IDE/editor you like.

3. I tried to compile it with Visual Studio Code, but it was pretty complicated for a "non-C" coder (I'm from the "web world"). 
I used Rider from JetBrains for compiling.

4. Basically, I set a breakpoint at **Identification.cs** to see how the program defines my motherboard name. 
It was PROART B760-CREATOR D4. 
I added a constant to **Model.cs** and a 'case' for Identification.cs. 
Well, the code is pretty self-explanatory.

5. I copied and pasted similar code in **SuperIOHardware.cs** from another motherboard with the same chip.

6. I set another breakpoint, debugged, and reviewed the data.

7. Then, in **SuperIOHardware.cs** in the 'case' for my motherboard, I tried to match the info from the other tools (step 1).

8. Then I adjusted parameters, couple of "compile - run program - adjust - compile - run" steps and done.

--

So, you have to do quite the same way.
First of all, you can just add the name of your motherboard to  **Identification.cs** and **Model.cs**.
Then add a 'case' statement to **SuperIOHardware.cs** near my code.
After that, compile the project and check how the program works (run the compiled **LibreHardwareMonitor.exe** from **LibreHardwareMonitor\bin\Debug\net472**).
And if everything is ok, then use **LibreHardwareMonitorLib.dll** for Fan Control (just copy and paste it into the Fan Control folder).

Keep in mind that you have to replace **LibreHardwareMonitorLib.dll** in Fan Control with your compiled **LibreHardwareMonitorLib.dll** every time you update Fan Control.
Until this (or your) pull request is merged into the master branch, build it and wait for the author of Fan Control to use it in the project.
